File: RotateArray.java
class Solution {
    public void rotate(int[] nums, int k) {
        int[] numsC = new int[nums.length];
        k %= nums.length;
        for (int i = k; i < numsC.length; i++)
            numsC[i] = nums[i-k];
        for (int i = 1, j = k-1; i <= k; i++, j--) 
            numsC[j] = nums[nums.length - i];
        for (int i = 0; i < numsC.length; i++)
            nums[i] = numsC[i];
    }
}

File: RotateArray.py
def rotate(nums, k):
    # for i in range(0, k):
    #     temp = nums[len(nums) - 1]
    #     for j in range(len(nums) - 1, -1, -1):
    #         nums[j] = nums[j-1]
    #     nums[0] = temp
    last_arr = []
    for i in range(0, k):
        last_arr.append(nums.pop())
    last_arr.reverse()
    return last_arr

    
nums = [-1,-100,3,99]
k = 2
print(rotate(nums, k))
print(nums)